Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/ConceptCollectionProjectController.java

```diff
@@ -38,269 +29,98 @@
-			List<IProjectConceptCollection> conceptCollectionList = null;
-			try {
-				// TODO: listProjectConceptCollection() is to be changed according to mapper
-				conceptCollectionList = projectConceptCollectionManager.listProjectConceptCollection(projectid, userId);
-			} catch (QuadrigaStorageException e) {
-				throw new QuadrigaStorageException();
-			}
-			if(conceptCollectionList == null){
-				logger.info("Concept Collection list is empty buddy");
-			}
-			model.addAttribute("conceptCollectionList", conceptCollectionList);
-			IProject project = projectManager.getProjectDetails(projectid);
-			model.addAttribute("project", project);
-			model.addAttribute("projectid", projectid);
-			return "auth/workbench/project/conceptcollections";
-		} else {
-			for (int i = 0; i < values.length; i++) {
-				try {
-					projectConceptCollectionManager.deleteProjectConceptCollection(projectid, userId, values[i]);
-				} catch (QuadrigaStorageException e) {
-					throw new QuadrigaStorageException();
-				}
-				if(!msg.equals("")){
-					flag=1;
-				}
-			}
-		}
-		if(flag==0){
-			model.addAttribute("deletesuccess", 1);
-		}else{
-			model.addAttribute("deletesuccess", 0);
-		}
-		List<IProjectConceptCollection> conceptCollectionList = null;
-		try {
-			// TODO: listProjectConceptCollection() is to be changed according to mapper
-			conceptCollectionList = projectConceptCollectionManager.listProjectConceptCollection(projectid, userId);
-		} catch (QuadrigaStorageException e) {
-			throw new QuadrigaStorageException();
-		}
-		if(conceptCollectionList == null){
-			logger.info("Dictionary list is empty buddy");
-		}
-		model.addAttribute("conceptCollectionList", conceptCollectionList);
-		IProject project = projectManager.getProjectDetails(projectid);
-		model.addAttribute("project", project);
-		model.addAttribute("projectid", projectid);
-		return "auth/workbench/project/conceptcollections";
-	}
+    @Autowired
+    IConceptCollectionManager conceptCollectionManager;
+
+    @Autowired
+    IRetrieveProjectManager projectManager;
+
+    @Autowired
+    private IProjectConceptCollectionManager projectConceptCollectionManager;
+
+    @RequestMapping(value = "auth/workbench/{projectid}/conceptcollections", method = RequestMethod.GET)
+    public String listProjectConceptCollection(@PathVariable("projectid") String projectid, Model model,
+            Principal principal) throws QuadrigaStorageException {
+        String userId = principal.getName();
+        List<IProjectConceptCollection> projectConceptCollectionList = projectConceptCollectionManager
+                .listProjectConceptCollection(projectid, userId);
+        model.addAttribute("projectConceptCollectionList", projectConceptCollectionList);
+        IProject project = projectManager.getProjectDetails(projectid);
+        model.addAttribute("project", project);
+        model.addAttribute("projectid", projectid);
+        return "auth/workbench/project/conceptcollections";
+    }

```

Comment:
Since you're add it. Each page/action should have its own controller:

- one for listing concept collections of a project
- one for adding concept collections to projects
- one for deleting concept collections from projects.